### DISCUSSION TITLE
eframe: Am I able to create/use UI functions in the main() function?

### DISCUSSION THREAD

**Discussion by Raj2032** [emilk/egui#1128] [Q&A]
Using the `eframe` framework, I wanted to ask you am I able to draw/create UI elements such as buttons, sliders, checkboxes etc inside the `main()` function rather than creating it in `impl` and running it in the `main()` function?

![image](https://user-images.githubusercontent.com/29293830/149777786-2974a612-fd18-4898-978e-61b198a2ab21.png)

As I was looking at some tutorial and it sounds like you can't do all the definitions in the `main()`?

**Comment by emilk** [comment]
if you are using `eframe` you need to use the `epi::App` trait.

If you want to use `egui` from your `fn main`, take a look at `egui-macroquad`: https://crates.io/crates/egui-macroquad

**Reply by Raj2032** [reply]
If I used egui-macroquad, can I make normal GUI applications?
